.github/workflows/code_quality.yml

```diff
@@ -0,0 +1,46 @@
+name: Code Quality Check
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  code-quality:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Set up PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '8.3'
+
+    - name: Configure Magento Auth for Composer
+      env:
+        COMPOSER_AUTH_USERNAME: ${{ secrets.COMPOSER_AUTH_USERNAME }}
+        COMPOSER_AUTH_PASSWORD: ${{ secrets.COMPOSER_AUTH_PASSWORD }}
+      run: |
+        composer config --global http-basic.repo.magento.com $COMPOSER_AUTH_USERNAME $COMPOSER_AUTH_PASSWORD
+
+    - name: Install dependencies for CI
+      run: |
+        COMPOSER=composer-ci.json composer install --no-interaction --prefer-dist
+
+    - name: Run PHP Code Sniffer (phpcs)
+      id: phpcs
+      run: |
+        vendor/bin/phpcs . --standard=Magento2 --ignore=vendor --severity=10
```

Comment:
It would have been better with a phpcs.xml file, but well, this way should be okay too.

Reply:
Oh, I can try to do it that if you prefer that way

Reply:
It's done now